### PR TITLE
Feature-Own-Enumerables

### PR DESCRIPTION
**Pull request by maximoortelli**
### **Description**

_Own-enumerable feature branch, implementing a custom enumerable module within a Ruby class. Create a class MyList and a module MyEnumerable. Implement a subset of Enumerable's functionality, including the methods #all? , #any?, and #filter._

### **General Requirements:**

- [x] Ensure no linter errors.

- [x] Follow the correct GitHub Flow.

- [x] Document work in a professional manner.

### **Ruby Requirements:**

- [x] Adhere to the list of best practices for Ruby.

- [x] Check official documentation for the methods in Enumerable.

- [x] Create a module MyEnumerable that implements the following methods with the same functionality as methods in Enumerable:

- #all?

- #any?

- #filter

- Separate each class and module into individual .rb files.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class MyList
+  include MyEnumerable
+
+  def initialize(*args)
+    @list = args
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (ShadyShawkat) [inline my_list.rb]: - [x] Kindly make sure to implement this method to yield list members from the list. You've done so but using `each`. While this is correct, the purpose of this method is to learn about `yield` and how it works in a method. You can do something like the following to replicate the `each` behavior without actually using it.

```ruby
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @maximoortelli, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (maximoortelli) [reply my_list.rb]: Done, thank you 👍😃

**Review by Munsa1 — APPROVED** [review]
Hi @maximoortelli @Ademola-coding, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 


## Optional suggestions
- [ ] Next time kindly check the flow to be used for the project on the project requirements. You were supposed to use GitHub Flow instead of git-flow.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**